subsys/suit/platform/sdfw/src/suit_plat_authenticate.c

```diff
@@ -10,18 +10,61 @@
 #include <suit_plat_decode_util.h>
 #include <suit_plat_component_compatibility.h>
 #include <zephyr/logging/log.h>
-#ifdef CONFIG_SDFW_BUILTIN_KEYS
-#include <sdfw/sdfw_builtin_keys.h>
-#endif /* CONFIG_SDFW_BUILTIN_KEYS */
 
 LOG_MODULE_REGISTER(suit_plat_authenticate, CONFIG_SUIT_LOG_LEVEL);
 
+#ifdef MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
+#define MANIFEST_PUBKEY_NRF_TOP_GEN0	 0x4000BB00
+#define MANIFEST_PUBKEY_SYSCTRL_GEN0	 0x40082100
+#define MANIFEST_PUBKEY_OEM_ROOT_GEN0	 0x4000AA00
+#define MANIFEST_PUBKEY_APPLICATION_GEN0 0x40022100
+#define MANIFEST_PUBKEY_RADIO_GEN0	 0x40032100
+#define MANIFEST_PUBKEY_GEN_RANGE	 2
+
+/** @brief Convert int-based key ID into structure that holds both the ID and the key owner.
+ *
+ * @param[out]  key     Key structure to set.
+ * @param[in]   key_id  Int-based key ID.
+ *
+ * @returns SUIT_PLAT_SUCCESS  In case of successful conversion, SUIT_PLAT_ERR_INVAL otherwise.
+ */
+static suit_plat_err_t convert_int_key(mbedtls_svc_key_id_t *key, uint32_t key_id)
```

Comment:
You don't need to this. We want the source of request to be encoded as the owner, and then we will verify that the access is allowed. Secure domain is allowed to access all keys.

Reply:
Please agree on the common version with @osaether .
If that's the case, we can use a solution based on: https://github.com/nrfconnect/sdk-nrf/commit/4685d64f7026f261a25ee9102c8fc84dedcbd5b5